### PR TITLE
Fixing namespace label issues

### PR DESCRIPTION
**Pull request by vpinisetti-px**
**What this PR does / why we need it**:
To fix the namespace label issues

**Which issue(s) this PR fixes** (optional)
Closes # https://portworx.atlassian.net/browse/PB-5345

**Special notes for your reviewer**:

Vanilla: where the default labels are 3

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4556/consoleFull

OCP 4.13: where the default labels are 7

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4584/console

### CODE REVIEW

File: tests/backup/backup_restore_basic_test.go [hunk 1]
```diff
@@ -1742,9 +1749,9 @@ var _ = Describe("{AddMultipleNamespaceLabels}", func() {
 			log.InfoD("Verifying number of labels added to namespace")
 			fetchedLabelMap, err = Inst().S.GetNamespaceLabel(bkpNamespaces[0])
 			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the labels added to namespace %s", bkpNamespaces[0]))
-			length := len(fetchedLabelMap)
-			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching labels %v for namespace %v", length-3, bkpNamespaces[0]))
-			dash.VerifyFatal(length-3, desiredNumLabels, fmt.Sprintf("Verifying number of added labels to desired labels for namespace %v", bkpNamespaces[0]))
+			updatedLabelMapLength := len(fetchedLabelMap)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Length of updated labels for namespace %v : %v", updatedLabelMapLength, bkpNamespaces[0]))
```
Comment (kphalgun-px) [inline tests/backup/backup_restore_basic_test.go:1753]: How about?

```go
log.InfoD("Length of updated labels for namespace %s: %v", bkpNamespaces[0], updatedLabelMapLength)
```

File: tests/backup/backup_restore_basic_test.go [hunk 2]
```diff
@@ -1731,6 +1733,11 @@ var _ = Describe("{AddMultipleNamespaceLabels}", func() {
 			ValidateApplications(scheduledAppContexts)
 		})
 		Step("Adding labels to namespaces in multiple of 10 until 1000", func() {
+			log.InfoD("Verifying default labels in the namespace %s", bkpNamespaces[0])
+			defaultLabelMap, err = Inst().S.GetNamespaceLabel(bkpNamespaces[0])
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Default labels presented in the namespace %v : %v", bkpNamespaces[0], defaultLabelMap))
```
Comment (mkoppal-px) [inline tests/backup/backup_restore_basic_test.go:1738]: Use `log.FailOnError` here

File: tests/backup/backup_restore_basic_test.go [hunk 3]
```diff
@@ -1731,6 +1733,11 @@ var _ = Describe("{AddMultipleNamespaceLabels}", func() {
...
+			defaultLabelMapLength = len(defaultLabelMap)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Lenth of default labels in the namespace %v : %v", bkpNamespaces[0], defaultLabelMapLength))
```
Comment (mkoppal-px) [inline tests/backup/backup_restore_basic_test.go:1740]: nit: `Length`

**Comment by mkoppal-px** [comment]
This issue was mainly seen on OCP/IBM. Please run the tests against those platforms.

**Comment by vpinisetti-px** [comment]
> Do you really need a number of labels? Why can we fetch the list at start and then just add it to the desired labels and just match the two lists?

@kshithijiyer-px This is required as per test steps. Even after comparison two maps will end up with label count as per the test steps to verify all 1000 labels got applied or not and also I believe this is much simpler solution.

**Comment by kshithijiyer-px** [comment]
> > Do you really need a number of labels? Why can we fetch the list at start and then just add it to the desired labels and just match the two lists?
> 
> @kshithijiyer-px This is required as per test steps. Even after comparison two maps will end up with label count as per the test steps to verify all 1000 labels got applied or not and also I believe this is much simpler solution.

@vpinisetti-px The reason of asking to check if the 1000 labels are the same labels is that programmatically it's easy to check if 2 maps have the same value.

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**